test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -425,6 +426,93 @@ public function testGetParentClassWithMissingParent(): void
         $classInfo->getParentClass();
     }
 
+    public function testGetInterfaceNamesWithMissingInterfaceDefinitions(): void
+    {
+        $classInfo = (new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+            $this->astLocator,
+        )))->reflectClass(ClassWithMissingInterface::class);
+
+        $this->expectException(IdentifierNotFound::class);
+
+        self::assertNotNull($classInfo->getInterfaceNames());
+    }
+
+    public function testGetInterfacesWithMissingInterfaceDefinitions(): void
+    {
+        $classInfo = (new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+            $this->astLocator,
+        )))->reflectClass(ClassWithMissingInterface::class);
+
+        $this->expectException(IdentifierNotFound::class);
+
+        self::assertNotNull($classInfo->getInterfaces());
+    }
```

Comment:
these tests show, that the pre-PR existing API always throws and therefore does not allow to get the interface names, when the class-symbols are not autoloadable.